internal/session/error.go

```diff
@@ -8,12 +8,4 @@ var (
 	// ErrInvalidStateForOperation indicates the session's state does not allow the
 	// operation.
 	ErrInvalidStateForOperation = errors.New("state is invalid for operation")
```

Comment:
Why not refactor this and the dependent code in repository_session.go to use the new errors?

Reply:
Ah.. figured it out.. it's the protocol aware anti-pattern. Would be nice to fix this somehow.

Reply:
Yes, I am going to have a quick follow-up PR, I chatted with a couple of people and I think we can just nuke the GRPC status returns here, however, I did not want to sneak it in this PR.